Document connect and sign (#998)
MetaMask/metamask-docs: wallet/how-to/sign-data/connect-and-sign.md

---
description: Use MetaMask SDK to connect and sign in a single interaction.
sidebar_position: 1
---

> # Connect and sign

Reviewer: Connect and sign using the SDK?

> You can connect to MetaMask and sign data in a single interaction from your JavaScript or iOS dapp
> with [MetaMask SDK](../../concepts/sdk/index.md) installed.

Reviewer: ```suggestion
You can connect to MetaMask and sign data in a single interaction from your JavaScript or iOS dapp
using the [MetaMask SDK](../../concepts/sdk/index.md).
```

> The SDK provides a `connectAndSign` method, which is a streamlined approach for dapps to interact
> with MetaMask.

Reviewer: ```suggestion
The SDKs `connectAndSign` method provides a streamlined approach for dapps to interact
with MetaMask.
```

This method combines the [`eth_requestAccounts`] and [`personal_sign`] RPC methods, executing them sequentially.
`connectAndSign` takes one parameter, the message string to be signed, and passes the message and
the output of [`eth_requestAccounts`] directly to [`personal_sign`].

> This streamlined method improves dapp user experience, especially on mobile platforms, since it
> requests the user to connect to MetaMask and sign a message in a single interaction, requiring only
> one switch between the mobile dapp and MetaMask Mobile.

Reviewer: ```suggestion
This method enhances dapp user experience, especially on mobile platforms, by allowing users to
connect to MetaMask and sign a message in one interaction, requiring only one switch between the
mobile dapp and MetaMask Mobile.
```
